R/data.R

```diff
@@ -58,6 +58,8 @@ eia_data <- function(dir, data = NULL, facets = NULL,
 }
 
 .eia_data <- function(dir, data, facets, freq, start, end, sort, length, offset, tidy, key){
+  md <- eia_metadata(dir, TRUE, TRUE, key)
```

Comment:
Does this mean that every API request becomes a minimum of two API requests?

Reply:
Unfortunately, yes. I'm not sure how else having truly dynamic validation of input values could be accomplished - unless we precompiled within the package itself (maybe within `data-raw/`...?) all the available options for every API data endpoint.

Reply:
So, completely dynamic- typical exception handling around the data call, rather than an automatic preemptive request.

Reply:
The other thing to realize, is that it will only make two API calls on the first function call to that endpoint. After that initial call, that endpoint and its metadata will be cached, so only one API call is made from then on within the context of that API endpoint (`dir` entry).

Reply:
Oh no - I like this. Good suggestion.